op-challenger/flags/flags.go

```diff
@@ -284,17 +276,17 @@ var Flags []cli.Flag
 
 func CheckCannonFlags(ctx *cli.Context) error {
 	if !ctx.IsSet(flags.NetworkFlagName) &&
-		!(ctx.IsSet(CannonRollupConfigFlag.Name) && ctx.IsSet(CannonL2GenesisFlag.Name)) {
+		!(RollupConfigFlag.IsSet(ctx, types.TraceTypeCannon) && L2GenesisFlag.IsSet(ctx, types.TraceTypeCannon)) {
 		return fmt.Errorf("flag %v or %v and %v is required",
-			flags.NetworkFlagName, CannonRollupConfigFlag.Name, CannonL2GenesisFlag.Name)
+			flags.NetworkFlagName, RollupConfigFlag.DefaultName(), L2GenesisFlag.DefaultName())
```

Comment:
(optional) It looks like you could set either `--rollup-config` or `--cannon-rollup-config`.  Maybe we should have some method like `VMFlag.OptionallyTypedName(vmType)` that returns something like "[cannon-]rollup-config".  Just a thought - maybe too confusing:
```suggestion
			flags.NetworkFlagName, RollupConfigFlag.OptionallyTypedName(types.TraceTypeCannon), L2GenesisFlag.DefaultName())
```

Reply:
Yeah good idea. Done.